src/pages/Organisation/Organisation.tsx

```diff
@@ -500,136 +532,214 @@ export const Organisation: React.FC = () => {
-          !(formState.isDirty && (formState.isValid || !Object.keys(formState.errors).length) && validImage)
-        }
-        onClick={org.name ? updateOrgDetails : createOrgDetails}
-      >
-        {t(
-          org.name ? "profileTab.organisationSubTab.buttonLabels.updateOrgButtonLabel" : "profileTab.organisationSubTab.buttonLabels.createOrgButtonLabel"
-        )}
-      </Button>
+      <Box>
+        <Typography variant="h6">
+          {t("profileTab.organisationSubTab.orgAddressTitle")}
+        </Typography>
+      </Box>
+
+      <Box mb={3} mt={3}>
+        <TextField
+          className={classes.inputFields}
+          error={formState.errors.orgAddress}
+          fullWidth
+          label={t("profileTab.organisationSubTab.fieldLabels.orgAddressLabel")}
+          margin='dense'
+          name='orgAddress'
+          onChange={handleChange}
+          style={{
+            display: "block",
+            maxWidth: "100%",
+            width: "100%",
+          }}
+          type='text'
+          value={formState.values.orgAddress}
+          variant='outlined'
+        />
+      </Box>
+
+      <Grid container spacing={3}>
+        <Grid item md={2}>
+          <TextField
+            className={classes.inputFields}
+            fullWidth
+            label={t("profileTab.organisationSubTab.fieldLabels.orgPostalCodeLabel")}
+            margin='dense'
+            name='orgPostalCode'
+            onChange={handleChange}
+            type='text'
+            value={formState.values.orgPostalCode}
+            variant='outlined'
+          />
+        </Grid>
+
+        <Grid item md={4}>
+          <TextField
+            className={classes.inputFields}
+            fullWidth
+            label={t("profileTab.organisationSubTab.fieldLabels.orgCityLabel")}
+            margin='dense'
+            name='orgCity'
+            onChange={handleChange}
+            type='text'
+            value={formState.values.orgCity}
+            variant='outlined'
+          />
+        </Grid>
+
+        <Grid item md={6}>
+          {/* <TextField
+            className={classes.inputFields}
+            fullWidth
+            label={t("profileTab.organisationSubTab.fieldLabels.orgCountryLabel")}
+            margin='dense'
+            name='orgCountry'
+            onChange={handleChange}
+            type='text'
+            value={formState.values.orgCountry}
+            variant='outlined'
+          /> */}
+
+          <CountrySelector countrySelectionHandler={handleCountrySelection} selectedCountry={selectedCountry} />
+        </Grid>
+      </Grid>
+
+      <Box mt={3}>
+        <Button
+          color="primary"
+          variant="contained"
+          size="large"
+          disableElevation
+          disabled={
+            !(
+              formState.isDirty &&
+              (formState.isValid || !Object.keys(formState.errors).length) &&
+              validImage
+            ) && selectedCountry === org.address.country
```

Comment:
Won't this keep the button disabled if the country is not changed and the rest of the form does change?